src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case `addProperties`:
+        copyState = Object.assign({}, copyState, action.extraData);
```

Comment:
It seems to me you can skip the assignment of an empty object as the first argument of this function, skip rewriting copyState, and just type:
Object.assign(copyState, action.extraData); 
in this line.